### PR TITLE
Export comments of forum as csv

### PR DESCRIPTION
**Pull request by franciclo**
- [x]  Add date creation

### CODE REVIEW

File: lib/admin/admin-comments/csv.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    var replies = (comment.replies.length > 0) ? comment.replies : [{text: '', author: {fullName: '', email: ''}}]
+    return replies.map(function (reply, i) {
+      return [
+        comment.topicId.id,
+        comment.topicId.mediaTitle,
+        `${config.protocol}://${config.host}/consulta/${comment.topicId.id}`,
+
+        '"' + comment.id + '"',
+        '"' + comment.context + '"',
+        '"' + comment.category + '"',
```
Comment (mjlescano) [inline lib/admin/admin-comments/csv.js]: Comments doesn't have categories here :/

File: lib/api/comment/index.js [hunk 2]
```diff
@@ -17,6 +18,59 @@ var app = module.exports = express()
 
 app.use(accepts('application/json'))
 
+app.get('/comment/all', requireTopicId, function (req, res) {
```
Comment (mjlescano) [inline lib/api/comment/index.js]: There's no need for this endpoint at all.

File: lib/db-api/comment.js [hunk 3]
```diff
@@ -32,8 +32,35 @@ var fields = 'id firstName lastName fullName email profilePictureUrl'
...
+          return topic.id
+        })
+        Comment.find(
+          {'_id': {$in: topicsId}},
+          function (err, docs) {
+            fn(null, comments)
+          })
+      })
+    } else {
+      log('Unknown option in all method')
```
Comment (mjlescano) [inline lib/db-api/comment.js]: needs to call the callback, something like `fn(new Error('Unknown option in comment.all'))`

File: lib/api/comment/index.js [hunk 4]
```diff
@@ -1,76 +1,76 @@
...
-
-app.get('/comment/all', requireTopicId, function (req, res) {
-  log('Request /comment/all for %s', req.query.topicId)
-
-  var sort = ''
-  if (~['-score', '-createdAt', 'createdAt'].indexOf(req.query.sort)) {
-    sort = req.query.sort
-  } else {
-    sort = '-score'
+app.get('/comment/:forum/all.csv', restrict, getForum, privileges('canChangeTopics'), function (req, res) {
```
Comment (mjlescano) [inline lib/api/comment/index.js]: `forumId` should be passed as a Query param, like `/comment/all.csv?forumId=${id}`

File: lib/db-api/comment.js (hunk 3 above)
Comment (franciclo) [inline lib/db-api/comment.js]: done

**Comment by mjlescano** [comment]
Comments are beign exported with a weird spacing between rows, e.g.:

```csv
Topic ID,Topic Title,Topic URL,Comment ID,Comment Date,Comment Time,Comment Text,Comment Author Fullname,Reply ID,Reply Date,Reply Time,Reply Text,Reply Author Fullname
574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"574c9ccfb3000e1000128622","5/30/2016","5:04:31 PM","Test Comment 2","Matías Lescano","574c9cdab3000e1000128623","5/30/2016","5:04:42 PM","Test Reply","Matías Lescano",574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"574c9ccfb3000e1000128622","5/30/2016","5:04:31 PM","Test Comment 2","Matías Lescano","587fdbdfffb4d62db54f56f7","1/18/2017","6:19:27 PM","Respuesta!","Matías Lescano",574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"574c9ccfb3000e1000128622","5/30/2016","5:04:31 PM","Test Comment 2","Matías Lescano","587fdbe4ffb4d62db54f56f8","1/18/2017","6:19:32 PM","sarasa---","Matías Lescano"



574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"57e2978005177b0300655007","9/21/2016","11:21:52 AM","<img onerror='document.body.innerHTML='<b src=\'text-align:center; display:block; position: fixed; top: 0; bottom: 0; right: 0; left: 0; background-color: #000; color: #fff; font-size: 25px; line-height: 100%;\'>HACKED</b>'' src=b>","Matías Lescano","58081e27f1b6b803002a192c","10/19/2016","10:30:15 PM","asdfg","Matías Lescano",574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"57e2978005177b0300655007","9/21/2016","11:21:52 AM","<img onerror='document.body.innerHTML='<b src=\'text-align:center; display:block; position: fixed; top: 0; bottom: 0; right: 0; left: 0; background-color: #000; color: #fff; font-size: 25px; line-height: 100%;\'>HACKED</b>'' src=b>","Matías Lescano","587d232427ad7ac4d845d096","1/16/2017","4:46:44 PM","<script>console.log('hacked!')</script>","Matías Lescano"






574c9cb3b3000e1000128619,Test,http://localhost/consulta/574c9cb3b3000e1000128619,"587fdafeffb4d62db54f56f5","1/18/2017","6:15:42 PM","que tal?","Matías Lescano","587fdb2affb4d62db54f56f6","1/18/2017","6:16:26 PM","response","Matías Lescano"
```

**Comment by mjlescano** [comment]
- [] Add createdAt column with the DateTime on GMT-0
- [] For the Date use the format `YYYY/MM/DD`

**Pull request merged**